src/qt/buynamespage.cpp

```diff
@@ -86,20 +87,31 @@ QString BuyNamesPage::firstupdate(const QString &name, const std::optional<QStri
     LogPrintf ("wallet attempting name_firstupdate: name=%s\n", strName);
 
     UniValue params(UniValue::VOBJ);
-    params.pushKV ("name", strName);
+
+    valtype vtName = valtype (strName.begin (), strName.end ());
+    std::string hexName = EncodeName (vtName, NameEncoding::HEX);
```

Comment:
Seems this code has the same type-safety issue that you brought up in https://github.com/namecoin/namecoin-core/pull/482#discussion_r776625657 .  Would it be okay if I refactored this so that the string argument is in hex form rather than binary form?

Reply:
Yes sure.

Reply:
I decided to defer this refactor until the GUI is updated to handle hex data, and instead just added a conversion function as you requested.